### PR TITLE
[FLINK-9866] Allow passing command line arguments to standalone job

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

Allow passing program arguments to standalone job

## Brief change log

*(for example:)*
  - Pass parsed program arguments to StandloneJobClusterEntry
  - updated standalone-job.sh script
  - updated docker scripts

## Verifying this change

* extended org.apache.flink.container.entrypoint.StandaloneJobClusterEntryPointTest to take parameters into account

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / **docs** / JavaDocs / not documented)

### CODE REVIEW

File: flink-container/kubernetes/README.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ The files contain the following variables:
 
 - `${FLINK_IMAGE_NAME}`: Name of the image to use for the container
 - `${FLINK_JOB}`: Name of the Flink job to start (the user code jar must be included in the container image)
+- `${FLINK_JOB_ARGUMENS}`: Job specific command line arguments
```
Comment (yanghua) [inline flink-container/kubernetes/README.md]: Shall we give a example or more documentation to guide how to pass the command line arguments? for example the format, like "--arg val" or something else? because here are many formats such as "--key value", "-Dxxx=xx".

File: flink-dist/src/main/flink-bin/bin/standalone-job.sh [hunk 2]
```diff
@@ -56,7 +56,7 @@ if [[ $STARTSTOP == "start" ]] || [[ $STARTSTOP == "start-foreground" ]]; then
     export FLINK_ENV_JAVA_OPTS="${FLINK_ENV_JAVA_OPTS} ${FLINK_ENV_JAVA_OPTS_JM}"
 
     # Startup parameters
-    ARGS+=("--configDir" "${FLINK_CONF_DIR}")
+    ARGS=("--configDir" "${FLINK_CONF_DIR}" "${ARGS[@]}")
```
Comment (tillrohrmann) [inline flink-dist/src/main/flink-bin/bin/standalone-job.sh]: Why is this change necessary? `ARGS` should already contain the remaining command line arguments.

Comment (dawidwys) [reply flink-dist/src/main/flink-bin/bin/standalone-job.sh]: You are right, this is not necessary. I thought the program arguments must be passed last, but that's obviously not true.

File: flink-container/kubernetes/README.md (hunk 1 above)
Comment (dawidwys) [reply flink-container/kubernetes/README.md]: I removed it for the time being as in fact I haven't addresed the kubernetes part yet.

**Comment by dawidwys** [comment]
Merging this, when my travis gives green.

File: flink-dist/src/main/flink-bin/bin/standalone-job.sh (hunk 2 above)
Comment (tillrohrmann) [reply flink-dist/src/main/flink-bin/bin/standalone-job.sh]: No I think you're right and the additional command line arguments need to be passed last. Please remove the initial definition of `ARGS` if you fix it like this here.
